server/src/main/java/org/elasticsearch/index/mapper/SourceFieldMapper.java

```diff
@@ -340,24 +376,25 @@ public SourceFieldMapper build() {
     }
 
     private static SourceFieldMapper resolveSourceMode(final IndexMode indexMode, final Mode sourceMode, boolean enableRecoverySource) {
-        if (indexMode == IndexMode.STANDARD) {
-            return enableRecoverySource ? DEFAULT : DEFAULT_NO_RECOVERY_SOURCE;
-        }
-        final SourceFieldMapper syntheticWithoutRecoverySource = indexMode == IndexMode.TIME_SERIES
-            ? TSDB_DEFAULT_NO_RECOVERY_SOURCE
-            : LOGSDB_DEFAULT_NO_RECOVERY_SOURCE;
-        final SourceFieldMapper syntheticWithRecoverySource = indexMode == IndexMode.TIME_SERIES ? TSDB_DEFAULT : LOGSDB_DEFAULT;
-        final SourceFieldMapper storedWithoutRecoverySource = indexMode == IndexMode.TIME_SERIES
-            ? TSDB_DEFAULT_NO_RECOVERY_SOURCE_STORED
-            : LOGSDB_DEFAULT_NO_RECOVERY_SOURCE_STORED;
-        final SourceFieldMapper storedWithRecoverySource = indexMode == IndexMode.TIME_SERIES ? TSDB_DEFAULT_STORED : LOGSDB_DEFAULT_STORED;
-
-        return switch (sourceMode) {
-            case SYNTHETIC -> enableRecoverySource ? syntheticWithRecoverySource : syntheticWithoutRecoverySource;
-            case STORED -> enableRecoverySource ? storedWithRecoverySource : storedWithoutRecoverySource;
-            case DISABLED -> throw new IllegalArgumentException(
-                "_source can not be disabled in index using [" + indexMode + "] index mode"
-            );
+        return switch (indexMode) {
```

Comment:
Introduce handling of `synthetic`, `stored` and `disabled` for `standard` too.

Reply:
I will just replace this with an old-style switch to avoid later on an issue with the backport which requires a lower language level.